js/ice3x.js

```diff
@@ -123,10 +122,6 @@ module.exports = class ice3x extends Exchange {
     }
 
     async fetchMarkets (params = {}) {
-        if (!Object.keys (this.currencies).length) {
```

Comment:
Not so fatso:    

```
'XRP/BTC': {     limits: { amount: { min: undefined, max: undefined },
                               price: { min: undefined, max: undefined },
                                cost: { min: undefined, max: undefined }  },
                 precision: { amount: 8, price: 8 },
                 tierBased:    undefined,
                percentage:    undefined,
                     taker:    0.01,
                     maker:    0.01,
                        id:   "27",
                    symbol:   "XRP/BTC",
                      base:   "XRP",
                     quote:   "BTC",
                    baseId:   "15",
                   quoteId:   "1",
                    active:    true,
                      info: {          pair_id:    27,
                              currency_id_from:    15,
                                currency_id_to:    1,
                                       filters: {  price: { min: "0.00000001", step: "0.00000001" },
                                                  amount: { min: "0.00000001", step: "0.00000001" }  } } }  }

```

Check out `currency_id_from` and `currency_id_to` – those are numeric literals, so to get the currency code we have to load the currencies first in this case. In fact, we should probably always load the currencies first. However, this is a separate issue, and this is a different concern. I would suggest to restore the preloading of currencies for ice3x.